File: tools/serve-fixed.py
"""時刻を固定した Flask を立てる。差分比較のためだけに使う。"""

import sys
from datetime import datetime

from app import clock, web

now = datetime.fromisoformat(sys.argv[1])
clock.now = lambda: now
web.create_app(db_path=sys.argv[2]).run(host="127.0.0.1", port=int(sys.argv[3]), threaded=True)

File: tools/serve-fixed.mjs
/** 時刻を固定した Express を立てる。差分比較のためだけに使う。 */

import { createApp } from "../src/app.js";
import { clock } from "../src/clock.js";

const now = Number(process.argv[2]);
clock.now = () => now;
createApp({ dbPath: process.argv[3] }).listen(Number(process.argv[4]), "127.0.0.1");
